ios/ios/src/Persisted Objects/CMObject.m

```diff
@@ -151,15 +163,27 @@ - (void)save:(CMStoreObjectUploadCallback)callback;
     }
 }
 
-- (void)saveWithUser:(CMUser *)user callback:(CMStoreObjectUploadCallback)callback;
+- (void)saveAtUserLevel:(nullable CMStoreObjectUploadCallback)callback
 {
+    NSAssert(nil != [CMUser currentUser] && [CMUser currentUser].isLoggedIn, @"*** Error: Cannot save a user level object without a logged in user");
     NSAssert([self.store objectOwnershipLevel:self] != CMObjectOwnershipAppLevel, @"*** Error: Object %@ is already at the app-level. You cannot also save it to the user level. Make a copy of it with a new objectId to do this.", self);
+
+    if (nil == self.store.user) {
+        self.store.user = [CMUser currentUser];
+    }
+
     if ([self.store objectOwnershipLevel:self] == CMObjectOwnershipUndefinedLevel) {
         [self.store addUserObject:self];
     }
+
     [self save:callback];
 }
 
+- (void)saveWithUser:(CMUser *)user callback:(CMStoreObjectUploadCallback)callback;
```

Comment:
Good eye @spectorar. This method is being deprecated in this release. You are correct in saying the user parameter is ignored. It was always ignored, in fact. It is only possible to save an object to the user who is currently logged in, so there is no need to pass in a user. The replacement method is simply `saveAtUserLevel:`